lib/mozart_fetcher/fetcher.ex

```diff
@@ -12,14 +12,21 @@ defmodule MozartFetcher.Fetcher do
   def process(components) do
     ExMetrics.timeframe "function.timing.fetcher.process" do
       components
-      |> Enum.with_index
+      |> Enum.with_index()
       |> Enum.map(&Task.async(fn -> Component.fetch(&1) end))
-      |> Enum.map(&Task.await/1)
+      |> Enum.map(&Task.await(&1, max_timeout(components)))
       |> decorate_response
       |> Jason.encode!()
     end
   end
 
+  defp max_timeout(components) do
```

Comment:
wondering if this could go in the Timeout Parser module

Reply:
TimeoutParser.max(components.endpoint)

Reply:
so it could make this bit of code easily testable

Reply:
in any case I'd pass the componens subesect as it would make things a bit more generic

Reply:
👍 I agree

Reply:
I have moved the maximum timeout logic to the timeout parser.